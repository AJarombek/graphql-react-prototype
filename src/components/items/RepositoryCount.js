/**
 * RepositoryCount component which displays the number of repositories the user owns.
 * @author Andrew Jarombek
 * @since 4/15/2020
 */

import React, { useEffect, useState } from 'react';
import { getPersonalRepositories } from '../../datasource/GraphQL';

const RepositoryCount = () => {
  const [repoCount, setRepoCount] = useState(0);
  const [error, setError] = useState(null);

  useEffect(() => {
    async function getGraphQLResult() {
      const result = await getPersonalRepositories('AJarombek');

      if (result.data.data) {
        setRepoCount(result.data.data.user.repositories.totalCount);
        setError(null);
      } else {
        setError(result.data.errors[0].message);
      }
    }

    getGraphQLResult();
  }, []);

  return (
    <div className="items repository-count">
      {error
        ? (
          <div className="error">
            <h6>{error}</h6>
          </div>
        )
        : (
          <>
            <h3>Number of Repositories</h3>
            <h2>{repoCount}</h2>
          </>
        )}
    </div>
  );
};

export default RepositoryCount;
