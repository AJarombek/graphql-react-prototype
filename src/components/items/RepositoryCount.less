/**
 * Stylesheet for the RepositoryCount component.
 * Author: Andrew Jarombek
 * Date: 4/15/2020
 */

.repository-count {

}
