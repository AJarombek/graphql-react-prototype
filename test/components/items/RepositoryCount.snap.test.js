/**
 * Snapshot test for the RepositoryCount component.
 * @author Andrew Jarombek
 * @since 4/25/2020
 */

import React from 'react';
import RepositoryCount from '../../../src/components/items/RepositoryCount';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<RepositoryCount />).toJSON();
  expect(tree).toMatchSnapshot();
});
