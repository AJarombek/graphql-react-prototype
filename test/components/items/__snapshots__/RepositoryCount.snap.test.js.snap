// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="items repository-count"
>
  <h3>
    Number of Repositories
  </h3>
  <h2>
    0
  </h2>
</div>
`;
